modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,367 @@
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ * @return mixed $attr Attributes for the image markup.
+ */
+function dominant_color_update_attachment_image_attributes( $attr, $attachment ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	$has_transparency = isset( $image_meta['has_transparency'] ) ? $image_meta['has_transparency'] : true;
+
+	$extra_class = '';
+	if ( ! isset( $attr['style'] ) ) {
+		$attr['style'] = ' --has-transparency: ' . $has_transparency . '; ';
+	} else {
+		$attr['style'] .= ' --has-transparency: ' . $has_transparency . '; ';
+		$extra_class    = ' has-transparency ';
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+		$dominant_color = $image_meta['dominant_color'];
+	} else {
+		/**
+		 * Filters the default color to use when no dominant color is found.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $default_color The default color 'cccccc'.
+		 */
+		$dominant_color = apply_filters( 'dominant_color_default_color', 'cccccc' );
+	}
+
+	if ( ! empty( $dominant_color ) ) {
+
+		$attr['data-dominant-color'] = $dominant_color;
+
+		if ( empty( $attr['style'] ) ) {
+			$attr['style'] = '';
+		}
+		$attr['style'] .= '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+
+		$extra_class .= ( dominant_color_color_is_light( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+		if ( empty( $attr['class'] ) ) {
+			$attr['class'] = '';
+		}
+		if ( isset( $attr['class'] ) && ! array_intersect( explode( ' ', $attr['class'] ), explode( ' ', $extra_class ) ) ) {
+			$attr['class'] .= ' ' . $attr['class'];
+		}
+	}
+
+	return $attr;
+}
+add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_update_attachment_image_attributes', 10, 2 );
+
+/**
+ * Filter image tags in content to add the dominant color to the image.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context The context of the image.
+ * @param int    $attachment_id The attachment ID.
+ *
+ * @return string image tag
+ */
+function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $attachment_id ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
+	if ( ! is_array( $image_meta ) ) {
+
+		return $filtered_image;
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+		$dominant_color = $image_meta['dominant_color'];
+	} else {
+		/**
+		 * Filters the default color to use when no dominant color is found.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $default_color The default color 'cccccc'.
+		 */
+		$dominant_color = apply_filters( 'dominant_color_default_color', 'cccccc' );
+	}
+
+	if ( empty( $dominant_color ) ) {
```

Comment:
If the image has no transparency meta data set, we also need to bail early here.